sa/model.go

```diff
@@ -239,6 +239,19 @@ func registrationPbToModel(reg *corepb.Registration) (*regModel, error) {
 		return nil, err
 	}
 
+	// We don't want to write literal JSON "null" strings into the database if the
+	// list of contact addresses is empty. Replace any possibly-`nil` slice with
+	// an empty slice instead. We don't need to reference reg.ContactPresent,
+	// because we're going to write the whole object to the database anyway.
+	contact := reg.Contact
+	if len(contact) == 0 {
+		contact = []string{}
+	}
+	jsonContact, err := json.Marshal(contact)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We could make this more explicit and not rely on `json.Marshal`'s semantics:

```
var jsonContact string
if len(contact) == 0 {
  jsonContact = "[]"
} else {
  ... json.Marshal ...
}

Reply:
Yeah, this looks better. Done.